azext_edge/edge/providers/orchestration/resources/dataflows.py

```diff
@@ -66,6 +83,183 @@ def list(self, dataflow_profile_name: str, instance_name: str, resource_group_na
+            DataflowEndpointType.DATAEXPLORER.value,
+            DataflowEndpointType.DATALAKESTORAGE.value,
+            DataflowEndpointType.FABRICONELAKE.value,
+            DataflowEndpointType.LOCALSTORAGE.value,
+        ] and not schema_ref:
+            raise InvalidArgumentValueError(
+                f"'schemaRef' is required for dataflow due to destination endpoint '{destination_endpoint_type}' type."
+            )
+
+        # validate at least one of source and destination endpoint
+        # must have host with "aio-broker" that is MQTT endpoint
+        source_endpoint_host = source_endpoint_obj.get("properties", {}).get(
+            DATAFLOW_ENDPOINT_TYPE_SETTINGS[source_endpoint_type], {}).get("host", "")
+        destination_endpoint_host = desination_endpoint_obj.get("properties", {}).get(
+            DATAFLOW_ENDPOINT_TYPE_SETTINGS[destination_endpoint_type], {}).get("host", "")
+        is_source_local_mqtt = LOCAL_MQTT_HOST_PREFIX in source_endpoint_host and\
+            source_endpoint_type == DataflowEndpointType.MQTT.value
+        is_destination_local_mqtt = LOCAL_MQTT_HOST_PREFIX in destination_endpoint_host and\
+            destination_endpoint_type == DataflowEndpointType.MQTT.value
+        if not is_source_local_mqtt and not is_destination_local_mqtt:
+            raise InvalidArgumentValueError(
+                "Either source or destination endpoint must be Azure IoT Operations Local "
+                f"MQTT endpoint with host containing '{LOCAL_MQTT_HOST_PREFIX}'."
+            )
+
+    def _process_exist_endpoint(
+        self,
+        operations: list,
+        instance_name: str,
+        resource_group_name: str,
+        operation_type: str,
+    ) -> dict:
+        # get endpoint
+        operation = self._get_operation(
+            operations=operations,
+            operation_type=operation_type,
+        )
+
+        # get operation settings
+        operation_settings = operation.get(DATAFLOW_OPERATION_TYPE_SETTINGS[operation_type], {})
+        endpoint_name = operation_settings.get("endpointRef", "")
+
+        # call get_dataflow_endpoint
+        dataflow_endpoint = DataFlowEndpoints(self.cmd)
+        endpoint_obj = dataflow_endpoint.show(
+            name=endpoint_name,
+            instance_name=instance_name,
+            resource_group_name=resource_group_name,
+        )
+
+        if not endpoint_obj:
+            raise ResourceNotFoundError(
+                f"{operation_type} dataflow endpoint '{endpoint_name}' not found in instance '{instance_name}'. "
+                "Please provide a valid 'endpointRef' using --config-file."
+            )
+
+        return endpoint_obj
+
+    def _get_operation(
+        self,
+        operations: list,
+        operation_type: str,
+    ) -> dict:
+        operation = next(
+            (op for op in operations if op.get("operationType") == operation_type), {}
+        )
+        return operation
+
 
 class DataFlowEndpoints(Queryable):
```

Comment:
For \_\_init\_\_ here instances could be initialized first, then self.iotops_mgmt_client gets assigned the client nested under instances, which initializes with default subscription if none provided i.e.

```python
self.instances = Instances(cmd=cmd)
self.iotops_mgmt_client = self.instances.iotops_mgmt_client
```

Reply:
Similar optimization can be made for `DataFlowProfiles`